dropwizard-configuration/src/main/java/io/dropwizard/configuration/ResourceConfigurationSourceProvider.java

```diff
@@ -6,6 +6,11 @@
 public class ResourceConfigurationSourceProvider implements ConfigurationSourceProvider {
     @Override
     public InputStream open(String path) throws IOException {
+        InputStream result = getResourceAsStream(path);
+        return result == null && path.startsWith("/") ? getResourceAsStream(path.substring(1)) : result;
```

Comment:
Sorry, it is not about 

> convention of implicitly treating absolute resource paths as being rooted within the application's resources directory

it is about (as described in [Using Methods of java.lang.ClassLoader](https://docs.oracle.com/javase/8/docs/technotes/guides/lang/resources.html#classloader))

> The methods in ClassLoader use the given String as the name of the resource without applying any absolute/relative transformation (see the methods in Class). The name should not have a leading "/".